src/modules-load/modules-load.c

```diff
@@ -52,12 +53,30 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent) {
+static void *worker_thread(void *arg) {
+        int sock_fd = *(int *) arg;
+        char buffer[32];
```

Comment:
this size is a bit random. given that kmod names are ultimately just filenames (well, with some suffix applied, but let's ignore that), maybe size this to NAME_MAX